src/net/sourceforge/kolmafia/chat/ChatFormatter.java

```diff
@@ -359,30 +370,40 @@ public static final void removeHighlighting() {
       return;
     }
 
-    for (int i = 0; i < patterns.length; ++i) {
-      if (patterns[i].equals(selectedValue)) {
-        String settingString = StyledChatBuffer.removeHighlight(i);
+    removeHightlighting(selectedValue);
+  }
 
-        String oldSetting = Preferences.getString("highlightList");
-        int startIndex = oldSetting.indexOf(settingString);
-        int endIndex = startIndex + settingString.length();
+  public static Collection<String> getHighlights() {
+    String[] split = Preferences.getString("highlightList").split("\n");
+    List<String> highlights = new ArrayList<>();
 
-        StringBuffer newSetting = new StringBuffer();
+    for (int i = 0; i < split.length; i += 2) {
+      highlights.add(split[i] + "\n" + split[i + 1]);
+    }
 
-        if (startIndex != -1) {
-          newSetting.append(oldSetting, 0, startIndex);
+    return highlights;
+  }
 
-          if (endIndex < oldSetting.length()) {
-            newSetting.append(oldSetting.substring(endIndex));
-          }
-        }
+  public static void removeHightlighting(String selectedValue) {
+    selectedValue = selectedValue.toLowerCase();
 
-        String cleanString = newSetting.toString();
-        cleanString = ChatFormatter.MULTILINE_PATTERN.matcher(cleanString).replaceAll("\n");
-        cleanString = cleanString.trim();
+    String[] patterns = StyledChatBuffer.searchStrings.toArray(new String[0]);
 
-        Preferences.setString("highlightList", cleanString);
+    for (int i = 0; i < patterns.length; ++i) {
```

Comment:
I think `for (String pattern : StyledChatBuffer.searchStrings) {` would save having to make a new array.

I'm not certain, but the reason for the original "convert to array, loop over array" is, I think, that the `StyledChatBuffer.searchStrings` can change while being looped over and you could wind up past the end if you loop by indexes. But if you loop by iterators, I don't think that can happen. Not sure though.

Reply:
It would definitely throw an error.